lib/Builtins.cpp

```diff
@@ -14,171 +14,339 @@
 
 #include "Builtins.h"
 
+#include <unordered_map>
+#include <stdlib.h>
+
 using namespace llvm;
+using namespace clspv;
+
+////////////////////////////////////////////////////////////////////////////////
+// utilities for const char* map keys
+struct cstr_hash {
+  std::size_t operator()(const char *cstr) const {
+    std::size_t hash = 5381;
+    for ( ; *cstr != '\0' ; ++cstr)
+      hash = (hash * 33) + *cstr;
+    return hash;
+  }
+};
+struct cstr_eq {
+  bool operator()(const char* a, const char *b) const { return strcmp(a, b) == 0; }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+////  Convert Builtin function name to a Type enum
+////////////////////////////////////////////////////////////////////////////////
+static Builtins::EBuiltinType GetBuiltinType(const std::string& _name) {
```

Comment:
I prefer using an anonymous namespace block rather than scattering "static" in many places.